app/api/crud.py

```diff
@@ -8,6 +8,16 @@
 from . import utility as util
 
 
+def build_node_request_urls(node_urls: list, path: str) -> list:
```

Comment:
```suggestion
def build_node_request_urls(node_urls: list, base_path: str) -> list:
```
let's keep the name consistent

Reply:
To be clear, this is not the base path! It's the endpoint path, e.g. `/query`, `/assessments`, etc.

The base path is user-configured independently of the actual API endpoints, something like `/api` - and this would be part of the `node_url` itself. The base path would not be defined in the actual endpoint functions themselves.

Reply:
OK, fair enough, but the call signature just below reads:

```
build_node_request_urls(
            util.FEDERATION_NODES, attribute_base_path
        )
```
that is a bit confusing then I think

Reply:
Ah, I see! I think I originally named it that way because it refers to the 'base' of a given attribute router, e.g.:
`/assessments` is the base of the router (which itself returns all instances), and `/assessments/vocab` is a subpath of the router (returns all vocabulary terms), and so on for `/diagnoses`, `/pipelines`, etc.

So 'base path' is used contextually here, where the **app** base path is the root/base of the entire API, defined externally, and **attribute** base path is the root/base of a given attribute router.

However, if that's confusing, I can just rename it to `attribute_path` as long as we understand that attribute router subpaths (e.g., `/assessments/vocab`) return different data than a given attribute path (e.g., `/assessments`)!

Reply:
yeah, I think attribute_path or endpoint_path is better here. The main thing for me is that we need to keep the cognitive load of reading and understanding the code as low as we can. So whatever we can do for that, we should